src/appmake/mz.c

```diff
@@ -792,28 +792,26 @@ int mz_exec(char *target)
 
 
 		/* Write out the MZ file */
-		
+
 		/* ******** */
 		/*  HEADER  */
 		/* ******** */
-		
+
 		fputc(1,fpout);			/* MZ80K M/C progtam file type */
-		
+
 		/* Deal with the filename */
-		if (strlen(blockname) > 16 ) {
-			strncpy(name,blockname,16);
-		} else {
-			strcpy(name,blockname);
-			//strncat(name,"\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0",17-strlen(blockname));
-			strncat(name,"\15\15\15\15\15\15\15\15\15\15\15\15\15\15\15\15\15",17-strlen(blockname));
-		}
+
+        // To Do - It seems very weird to pad the filename with CR's (ascii char 15 octal)
+        memset(name, '\r', sizeof(name)-1);
```

Comment:
For true portability this should be '\15' rather than \r to cope with non-ascii host machines.